clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1581,6 +1581,23 @@ public void commitAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, Of
      * is invoked for the same partition more than once, the latest offset will be used on the next poll(). Note that
      * you may lose data if this API is arbitrarily used in the middle of consumption, to reset the fetch offsets
      *
+     * The next Consumer Record which will be retrieved when poll() is invoked will either have the offset specified or,
+     * by default, with {@link ConsumerConfig AUTO_RESET_CONFIG} set to "latest"
+     * a higher numbered offset. A higher numbered offset will be returned if there is no consumer record with the offset
+     * specified but there is one with a higher offset.
+     * seek(0) is equivalent to seek to beginning for a topic with beginning offset 0.
+     * seekToEnd() is equivalent to seeking to the highest known offset + 1.
+     *
+     * Seeking past the end of the highest known offset means an invalid offset is reached.
+     * Invalid offset behaviour is controlled by
+     * the {@link ConsumerConfig AUTO_RESET_CONFIG} property. If this is set to "earliest", the next poll() will seek to the beginning
+     * before returning a record. If it is set to "latest", it will seek to the last known record (similar to seekToEnd()).
```

Comment:
seek will happen right after you call `seek`, it doesn't happen after calling `poll`. How about this:
`If this is set to "earliest", the next poll will return records from the starting offset. `